app/controllers/matchings_controller.rb

```diff
@@ -54,6 +58,111 @@ def destroy
     end
   end
 
+  def category
+    return "person_to_time"
+  end
+
+  def global_settings
+    return {"minutes": @project.duration}
+  end
+
+  def people
+    all_participants_emails = Participant.where(project_id: @project.id).pluck(:email)
+
+    people = []
+    for email in all_participants_emails
+      row = {"name": email, "match_degree": 1}
+      people.push(row)
+    end
+
+    return people
+  end
+
+  def timeslots
+    all_project_times = ProjectTime.where(project_id: @project.id).pluck(:date_time)
+
+    timeslots = []
+    for timeslot in all_project_times
+      timeslot_formatted = timeslot.utc.strftime('%Y-%m-%d %H:%M %p')
+      row = {"timestamp": timeslot_formatted, "tracks": 1}
+      timeslots.push(row)
+    end
+
+    return timeslots
+  end
+
+  def preferences
+    all_participants_ids = Participant.where(project: @project.id).pluck(:id)
+    all_project_time_ids = ProjectTime.where(project_id: @project.id).pluck(:id)
+
+    preferences = []
+    all_participants_ids.each do |participant_id|
+      all_project_time_ids.each do |project_time_id|
+        person = Participant.find_by(project_id: @project.id)['email']
+        timeslot = ProjectTime.find_by(project_id: @project.id)['date_time']
+        timeslot_formatted = timeslot.utc.strftime('%Y-%m-%d %H:%M %p')
+        rank = Ranking.find_by(participant_id: participant_id, project_time_id: project_time_id)['rank']
+        row = {"person_name": person, "timeslot": timeslot_formatted, "rank": rank}
+        preferences.push(row)
+        end
+    end
+    return preferences
+  end
+
+  def all_submitted_preferences?
```

Comment:
This works fine, except I noticed that if there are no participants or no times, this directly returns true, and we see that everyone has submitted preferences and matching is allowed. Should add a warning that times are empty or participants are empty.